app/pods/publications/publication/translations/route.js

```diff
@@ -1,16 +1,86 @@
 import Route from '@ember/routing/route';
+import { action } from '@ember/object';
+import { PAGE_SIZE } from 'frontend-kaleidos/config/config';
+import { tracked } from '@glimmer/tracking';
 
+// use:
+// - isPieceDeletable property
+// - file property avoids error when piece (and file) are deleted
+export class TranslationRow {
+  @tracked piece;
+  @tracked date;
+  @tracked requestActivity;
+
+  // no async constructor() in JS
+  static async create(piece,requestActivity) {
+    const row = new TranslationRow();
+
+    if (piece){
+      row.piece = piece;
+      row.date = piece.created;
+    }
+    if (requestActivity) {
+      row.requestActivity = requestActivity;
+      row.date = requestActivity.startDate;
+    }
+    return row;
+  }
+
+  get isPieceDeletable() {
+    // can be translation or publication related
+    const isUsedInRequest = this.requestActivitiesUsedBy.length > 0;
+    // receivedDate is set if and only if it is a received pieced
+    const isReceived = !!this.piece.receivedDate;
+    const isUsed = isUsedInRequest || isReceived;
+    return !isUsed;
+  }
+}
 export default class PublicationsPublicationTranslationsRoute extends Route {
-  model() {
-    return this.modelFor('publications.publication').translationSubcase;
+  async model() {
+    this.translationSubcase = await this.modelFor('publications.publication').translationSubcase;
+
+    let requestActivities = await this.store.query('request-activity',
+      {
+        'filter[translation-subcase][:id:]': this.translationSubcase.id,
+        include: 'translation-activity,email,used-pieces,used-pieces.file',
+        sort: '-start-date',
+      }
+    );
+    let requestRows = await Promise.all(requestActivities.map((request) => TranslationRow.create(null,request)));
+
+    // Received translation documents generated by a translation-activity
+    const pieces = await this.store.query('piece', {
+      'filter[translation-activity-generated-by][subcase][:id:]': this.translationSubcase.id,
+      include: [
+        'file',
+        'publication-subcase-source-for',
+      ].join(','),
+      'page[size]': PAGE_SIZE.PUBLICATION_FLOW_PIECES,
+    });
+
+    const pieceRows = await Promise.all(pieces.map((piece) => TranslationRow.create(piece,null)));
+
+    let translationRows= [pieceRows,requestRows];
+    translationRows = translationRows.flatMap((row) => row.toArray())
+    translationRows = translationRows.sortBy('date').reverseObjects();
+
+    return translationRows;
```

Comment:
Deze model hook is niet logisch. Als je naar het design kijkt, bevat het scherm een tijdslijn van aanvraag-activiteiten en vertaal-activiteiten. Dat is dan ook wat ik zou verwachten in de model-hook, maar het is hier een mix van aanvraag-activiteit en pieces, die dan nog eens gewrapped zijn in een custom class.